src/pages/index.js

```diff
@@ -146,24 +202,107 @@ function closeModal(modal) {
   document.removeEventListener("keydown", handleEscapePress);
 }
 
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Delete", "Deleting...");
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitBtn.textContent = "Delete";
+    });
+}
+
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileNname.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editProfileModal);
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileNname.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editProfileModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitBtn.textContent = "Save";
+    });
 }
 
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
-  const inputValues = {
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  const cardData = {
     name: addCardNameInput.value,
     link: addCardLinkInput.value,
   };
-  const cardElement = getCardElement(inputValues);
-  cardList.prepend(cardElement);
-  addCardForm.reset();
-  closeModal(cardModal);
-  disableButton(submitButton, settings);
+  api
+    .addCard(cardData)
+    .then((newCard) => {
+      const cardElement = getCardElement(newCard);
+      cardList.prepend(cardElement);
+      addCardForm.reset();
+      closeModal(cardModal);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      disableButton(submitButton, settings);
+      submitBtn.textContent = "Save";
```

Comment:
` submitBtn.textContent = "Save";`

Please, reuse the `setButtonText` function for setting the button text


```js
    .finally(() => {
      setButtonText(submitBtn, false);
    });

```